pkg/artifactory/resource/security/resource_artifactory_scoped_token.go

```diff
@@ -8,371 +8,474 @@ import (
-			Computed: true,
-		},
-		"expiry": {
-			Type:     schema.TypeInt,
-			Computed: true,
-		},
-		"issued_at": {
-			Type:     schema.TypeInt,
-			Computed: true,
-		},
-		"issuer": {
-			Type:     schema.TypeString,
-			Computed: true,
 		},
 	}
+}
 
-	var unpackAccessTokenPostRequest = func(data *schema.ResourceData) (*AccessTokenPostRequest, error) {
-		d := &utilsdk.ResourceData{ResourceData: data}
+var serviceTypesScopedToken = []string{"jfrt", "jfxr", "jfpip", "jfds", "jfmc", "jfac", "jfevt", "jfmd", "jfcon"}
 
-		scopes := d.GetSet("scopes")
-		scopesString := strings.Join(scopes, " ") // Join slice into space-separated string
-		if len(scopesString) > 500 {
-			return nil, fmt.Errorf("total combined length of scopes field exceeds 500 characters: %s", scopesString)
-		}
+func (r *ArtifactoryScopedTokenResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+	r.ProviderData = req.ProviderData.(utilsdk.ProvderMetadata)
+}
 
-		audiences := d.GetSet("audiences")
-		audiencesString := strings.Join(audiences, " ") // Join slice into space-separated string
-		if len(audiencesString) > 255 {
-			return nil, fmt.Errorf("total combined length of audiences field exceeds 255 characters: %s", audiencesString)
-		}
+func (r *ArtifactoryScopedTokenResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *ArtifactoryScopedTokenResourceModel
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
-		accessToken := AccessTokenPostRequest{
-			Username:              d.GetString("username", false),
-			Scope:                 scopesString,
-			ExpiresIn:             d.GetInt("expires_in", false),
-			Refreshable:           d.GetBool("refreshable", false),
-			Description:           d.GetString("description", false),
-			Audience:              audiencesString,
-			IncludeReferenceToken: d.GetBool("include_reference_token", false),
-		}
+	scopes := []string{}
+	if !data.Scopes.IsNull() {
+		scopes = utilfw.StringSetToStrings(data.Scopes)
+	}
+	scopesString := strings.Join(scopes, " ") // Join slice into space-separated string
+	if len(scopesString) > 500 {
+		resp.Diagnostics.AddError(
+			"Scopes length exceeds 500 characters",
+			"total combined length of scopes field exceeds 500 characters:"+scopesString,
+		)
+		return
+	}
 
-		return &accessToken, nil
+	audiences := []string{}
+	if !data.Audiences.IsNull() {
+		audiences = utilfw.StringSetToStrings(data.Audiences)
+	}
+	audiencesString := strings.Join(audiences, " ") // Join slice into space-separated string
+	if len(audiencesString) > 255 {
+		resp.Diagnostics.AddError(
+			"Audiences length exceeds 255 characters",
+			"total combined length of audiences field exceeds 500 characters:"+scopesString,
```

Comment:
We should be using `len(audiencesString)` here instead of hardcoded '500' right?

Reply:
Copy/paste issue, fixed.